src/repository/models.py

```diff
@@ -623,6 +632,52 @@ def local_url(self):
 
         return url
 
+    def create_article(self, journal, workflow_stage, journal_license, journal_section, force=False):
+        """
+        Creates an article in a given journal and workflow stage.
+        """
+        if not self.article or force:
+            # create base article
+            article = submission_models.Article.objects.create(
+                journal=journal,
+                owner=self.owner,
+                title=self.title,
+                abstract=self.abstract,
+                license=journal_license,
+                section=journal_section,
+                date_submitted=timezone.now(),
+                comments_editor='Submitted from {}'.format(self.repository.name),
+                stage=workflow_stage,
+            )
+
+            # copy authors to submission
+            for preprint_author in self.preprintauthor_set.all():
```

Comment:
Do we not also need to `article.authors.add(preprint_author)` here?

Reply:
👍🏻